modules/core/04-channel/keeper/upgrade.go

```diff
@@ -299,6 +308,10 @@ func (k Keeper) ChanUpgradeAck(
 		return errorsmod.Wrapf(types.ErrUpgradeNotFound, "failed to retrieve channel upgrade: port ID (%s) channel ID (%s)", portID, channelID)
 	}
 
+	if err := k.checkForUpgradeSequences(ctx, portID, channelID, channel, counterpartyChannel.UpgradeSequence); err != nil {
+		return err
+	}
```

Comment:
Do we actually need this in `ChanUpgradeAck`? I see that the spec does not have this check, and I guess that if sequences would be different, the upgrade handshake would have already been stopped in `ChanUpgradeTry`?

Reply:
this logic was happening in `startFlushUpgradeHandshake` so was already happening in `ChanUpgradeAck`. The current code should be equivalent to that. I think it is probably safe to remove it though since it does happen in the try step!